src/app/(auth)/onboarding/page.tsx

```diff
@@ -0,0 +1,14 @@
+import { usersService } from "@/services/users-service";
+import { auth, redirectToSignIn } from "@clerk/nextjs";
+import { redirect } from "next/navigation";
+
+export default async function OnboardingRedirect() {
+  const { userId: clerkId } = auth();
+  if (!clerkId) return redirectToSignIn();
+
+  // fetch guarantees upsert immediately after sign up
+  await usersService.getUserByClerkId(clerkId);
+
+  const profileComplete = await usersService.profileComplete(clerkId);
+  return profileComplete ? redirect("/") : redirect("/profile/edit");
```

Comment:
Wouldn't the profile always be incomplete when the account has just been created? Or is this to navigate people back home if they accidentally go to the onboarding page

Reply:
Prevent accidental navigation